my-app/src/app/core/services/course/course.service.ts

```diff
@@ -1,49 +1,54 @@
+import { HttpClient } from '@angular/common/http';
 import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { finalize } from 'rxjs/operators';
 import { Course } from 'src/app/core/services/course/model/course';
-import { DateHelper } from 'src/app/shared/utils/date-helper';
+import { LoadingService } from '../loading/loading.service';
 
 @Injectable({
   providedIn: 'root'
 })
 export class CourseService {
 
+  private BASE_URL: string = 'http://localhost:3004/courses';
+
   public courses: Course[] = [];
 
-  constructor(private dateHelper: DateHelper) {
-    this.init();
-  }
+  constructor(private http: HttpClient, private loadingServce: LoadingService) {}
 
-  public addCourse(course: Course): void {
-    this.courses.push(course);
+  public search(text: string): Observable<Course[]> {
+    return this.http.get<Course[]>(`${this.BASE_URL}?textFragment=${text}`);
   }
 
-  public getList(): Course[] {
-    return this.courses.slice();
+  public getCourses(count: number): Observable<Course[]> {
+    this.loadingServce.loading$.next(true);
```

Comment:
probably, you could trigger loading in interceptor as far as all requests should trigger loading